runtime/src/qio/qio_popen.c

```diff
@@ -33,9 +33,46 @@
 #include <unistd.h>
 #include <spawn.h>
 
+// We need to be able to call malloc, free, etc.
+#include "chpl-mem-no-warning-macros.h"
+
 // current environment variables
 extern char** environ;
 
+/* Helper routines for allocating out of the C
+   heap instead of the Chapel one. That is necessary
+   since the Chapel heap might not be available to
+   a forked process, and we need args for exec
+   (command, arguments, environment) to be available
+   after a fork.
+ */
+
+const char* qio_spawn_strdup(const char* str)
+{
+  size_t len = strlen(str);
+  char* ret = malloc(len + 1);
+  // note: also copies '\0' at end of string.
+  memcpy(ret, str, len + 1);
+  return ret;
+}
+
+const char** qio_spawn_allocate_args(size_t count)
```

Comment:
These aren't necessarily args; rather it's a general vector of pointers. You might rename it as qio_spawn_allocate_ptrvec therefore, and similarly for the free() one.